File: backend/app/schemas/blog.py
"""博客前台聚合接口的 Pydantic 模型（§5.5）。"""
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class FeedQuery(BaseModel):
    sort: str = Field("latest", pattern="^(latest|hot)$")
    page: int = Field(1, ge=1)
    page_size: int = Field(10, ge=1, le=100)


class BlogArticleItem(BaseModel):
    id: int
    title: str
    summary: str
    cover_image: str
    view_count: int
    like_count: int
    published_at: Optional[datetime]
    created_at: datetime


class CategoryStat(BaseModel):
    id: int
    name: str
    article_count: int


class TagStat(BaseModel):
    id: int
    name: str
    use_count: int


class ArchiveItem(BaseModel):
    year: int
    month: int
    count: int

File: frontend/src/types/blog.ts
import type { Category, Tag } from './api'
import type { Article } from './article'

export interface BlogFeedParams {
  page?: number
  page_size?: number
  sort?: 'latest' | 'hot'
  /** 按分类筛选（博客分类浏览页使用） */
  category_id?: number
  /** 按标签名筛选（博客标签浏览页使用） */
  tag?: string
}

export interface HotArticle {
  id: number
  title: string
  summary?: string
  cover_image?: string
  view_count: number
  like_count: number
  published_at?: string | null
}

export interface CategoryStat extends Category {
  article_count?: number
}

export interface TagStat extends Tag {
  count?: number
}

export interface ArchiveItem {
  year: number
  month: number
  count: number
}

export interface RelatedArticle {
  id: number
  title: string
  summary?: string
  view_count: number
  like_count: number
}
